lib/src/repository.rs

```diff
@@ -99,72 +86,56 @@ pub trait RepositoryExt {
 }
 
 impl RepositoryExt for Repository {
-    fn get_issue_heads(&self, issue: Oid) -> Result<References> {
-        let glob = format!("**/dit/{}/head", issue);
-        self.references_glob(&glob)
-            .chain_err(|| EK::WrappedGitError)
-    }
-
-    fn get_local_issue_head(&self, issue: Oid) -> Result<Reference> {
-        let glob = format!("refs/dit/{}/head", issue);
-        self.references_glob(&glob)
-            .chain_err(|| EK::WrappedGitError)?
-            .next()
-            .ok_or_else(|| Error::from_kind(EK::CannotFindIssueHead(issue)))
-            .and_then(|reference| reference.chain_err(|| EK::WrappedGitError))
-    }
-
-    fn get_issue_leaves(&self, issue: Oid) -> Result<References> {
-        let glob = format!("**/dit/{}/leaves/*", issue);
-        self.references_glob(&glob)
-            .chain_err(|| EK::WrappedGitError)
-    }
-
-    fn get_issue_refs(&self, issue: Oid) -> Result<References> {
-        let glob = format!("refs/dit/{}/**", issue);
-        self.references_glob(&glob)
-            .chain_err(|| EK::WrappedGitError)
+    fn find_issue(&self, id: Oid) -> Result<Issue> {
+        let retval = Issue::new(self, id);
+
+        // make sure the id refers to an issue by checking whether an associated
+        // head reference exists
+        if retval.heads()?.next().is_some() {
+            Ok(retval)
+        } else {
+            Err(Error::from_kind(EK::CannotFindIssueHead(id)))
+        }
     }
 
-    fn get_issue_revwalk(&self, issue: Oid) -> Result<Revwalk> {
-        let glob = format!("**/dit/{}/**", issue);
-        self.revwalk()
-            .and_then(|mut revwalk| {
-                revwalk.push_glob(glob.as_ref())?;
-                revwalk.simplify_first_parent();
-                revwalk.set_sorting(git2::SORT_TOPOLOGICAL);
-                Ok(revwalk)
-            })
-            .chain_err(|| EK::WrappedGitError)
+    fn issue_by_head_ref(&self, head_ref: &git2::Reference) -> Result<Issue> {
+        head_ref.name()
+                .and_then(|name| if name.ends_with("/head") {Some(name)} else {None})
```

Comment:
Don't do these if-else-oneliners. Nobody can read this without getting insane. Especially not without proper whitespace.